### PR TITLE
Cleanup for accounting module and docker-compose file

### PR DESCRIPTION
**Pull request by mohsenone**

### CODE REVIEW

**Review by PouryaMansouri — APPROVED** [review]

File: cart/api.py [hunk 1]
```diff
@@ -126,23 +181,28 @@ def get_object(self):
...
-        except CartItem.DoesNotExist:
-            raise ValidationError({'error': ['محصول با این شناسه در سبد خرید شما وجود ندارد']})
+        except CartItem.DoesNotExist as ex:
+            raise ValidationError(
+                {'error': ['محصول با این شناسه در سبد خرید شما وجود ندارد']}
+            ) from ex
 
     def create(self, request, *args, **kwargs):
-        cart = self.get_cart()
+        cart = Cart.get_cart(request)
```
Comment (likecodingloveproblems) [inline cart/api.py:190]: If these two have the same functionality why do we have both of them?

**Review by likecodingloveproblems — COMMENTED** [review]
This is really a huge PR, I can't review all changes but I do my best.
thanks

**Review by likecodingloveproblems — COMMENTED** [review]
It's really a dangerous one, It's buggy

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+version: "3.7"
```
Comment (likecodingloveproblems) [inline docker-compose.yml:1]: It's buggy

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+services:
+  web:
+    build: ./app/
```
Comment (likecodingloveproblems) [inline docker-compose.yml:5]:
```suggestion
    build: ./
```

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    container_name: web
+    restart: unless-stopped
+    command: python manage.py runserver 0.0.0.0:8000
+    volumes:
+      - media_data:/usr/src/app/media
+    expose:
+      - "8000"
+    env_file:
+      - .env
```
Comment (likecodingloveproblems) [inline docker-compose.yml:14]: user must put a .env file in nakhll_backend directory

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on:
+      - postgres
+
+  postgres:
+    image: postgres:12.0-alpine
+    container_name: postgres
+    restart: unless-stopped
+    ports:
+      - "33659:5432"
```
Comment (likecodingloveproblems) [inline docker-compose.yml:23]: I don't take attention to this, this is a security issue, we export postgres port and port number is explicitly mentioned here, I doubt attacker must find our port!

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
+    env_file:
+      - .env
```
Comment (likecodingloveproblems) [inline docker-compose.yml:27]:
``` suggestion
      - .env

volumes:
  postgres_data:
  media_data:
  static_data:
```

Comment (likecodingloveproblems) [reply docker-compose.yml:27]: and also the user must create volume directories in nakhll_backend,
What's the use case of docker-compose.yml in nakhll_backend?

**Pull request merged**